File: Agent/Agent.SystemValue.Api/Types/ServiceEvent.cs
using Agent.SystemValue.Api.Types.Base;
using ProtoBuf;

namespace Agent.SystemValue.Api.Types
{
    [ProtoContract]
    public class ServiceEvent : ISystemValue, IEvent
    {
        [ProtoMember(1)] public string Executable { get; }

        [ProtoMember(2)] public ServiceEventType ServiceEventType { get; }

        [ProtoMember(10)] public string DeviceIdentifier { get; set; }

        public ServiceEvent()
        {
        }

        public ServiceEvent(string executable, ServiceEventType serviceEventType)
        {
            Executable = executable;
            ServiceEventType = serviceEventType;
        }

        public override string ToString() => $"ServiceEvent({Executable}, {ServiceEventType})";
    }

    [ProtoContract]
    public enum ServiceEventType
    {
        [ProtoEnum] Start = 1,
        [ProtoEnum] Stop = 2
    }
}

File: Watchdog/src/main/kotlin/systemValues/ServiceEvent.kt
package systemValues

import kotlinx.serialization.ExperimentalSerializationApi
import kotlinx.serialization.Serializable
import kotlinx.serialization.protobuf.ProtoNumber

@ExperimentalSerializationApi
@Serializable
data class ServiceEvent(
    @ProtoNumber(1) val executable: String,
    @ProtoNumber(2) val serviceEventType: ServiceEventType
) : SystemValue


@Serializable
enum class ServiceEventType {
    Start, Stop
}